src/main/java/androidx/widget/TextView.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.widget
+
+import android.graphics.drawable.Drawable
+import android.support.annotation.DrawableRes
+import android.support.annotation.RequiresApi
+import android.support.v4.content.ContextCompat
+import android.widget.TextView
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawables
+ */
+fun TextView.updateCompoundDrawables(
+    start: Drawable? = compoundDrawables[0],
```

Comment:
I wanted to collaborate to android-ktx on `TextView` and after digging a bit in the GitHub project, I found you PR. Nice work 👍 

The discussions you have here are really interesting and I actually ended up having the exact same questions. The feedback @romainguy gave is true. Using `compoundDrawables` 4 times to get the default value will possibly involve creating several arrays (one per call to `compoundDrawables`).

I ended up doing so:

```kotlin
private val drawableSentinel = DrawableContainer()

fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
        left: Drawable? = drawableSentinel,
        top: Drawable? = drawableSentinel,
        right: Drawable? = drawableSentinel,
        bottom: Drawable? = drawableSentinel
) {
    val drawables = lazy(LazyThreadSafetyMode.NONE) { compoundDrawables }
    setCompoundDrawablesWithIntrinsicBounds(
            if (left === drawableSentinel) drawables.value[0] else left,
            if (top === drawableSentinel) drawables.value[1] else top,
            if (right === drawableSentinel) drawables.value[2] else right,
            if (bottom === drawableSentinel) drawables.value[3] else bottom
    )
}
```

Reply:
BTW: I don't know which one is "lighter"

```kotlin
private val drawableSentinel = DrawableContainer()
```

or

```kotlin
private val drawableSentinel = ColorDrawable(Color.TRANSPARENT)
```

Reply:
I like this solution. @JakeWharton, does this look good to you?

Reply:
Thank you for the suggestion @cyrilmottier . I am happy to apply it if it is required by the maintainers.

Reply:
Tests passing with the proposed solution 👍 
```kotlin
private val drawableSentinel = DrawableContainer()
// ...
fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
    start: Drawable? = compoundDrawables[0],
    top: Drawable? = compoundDrawables[1],
    end: Drawable? = compoundDrawables[2],
    bottom: Drawable? = compoundDrawables[3]
) {
    val drawables = lazy(LazyThreadSafetyMode.NONE) { compoundDrawables }
    setCompoundDrawablesWithIntrinsicBounds(
        if (start === drawableSentinel) drawables.value[0] else start,
        if (top === drawableSentinel) drawables.value[1] else top,
        if (end === drawableSentinel) drawables.value[2] else end,
        if (bottom === drawableSentinel) drawables.value[3] else bottom
    )
}
```

Reply:
:+1:

Reply:
Hmmm I think you forgot to use the `drawableSentinel` in the default parameters.

Reply:
Quite! Corrected 😨

Reply:
Version that can be reused by other `Drawable?` methods?:
```kotlin
private val drawableSentinel = DrawableContainer()
// ...
private fun updateDrawables(
    start: Drawable?, top: Drawable?, end: Drawable?, bottom: Drawable?, srcArray: Array<Drawable>
): Array<Drawable?> {
    val drawables = lazy(LazyThreadSafetyMode.NONE) { srcArray }
    return arrayOfNulls<Drawable>(srcArray.size).apply {
        this[0] = if (start === drawableSentinel) drawables.value[0] else start
        this[1] = if (top === drawableSentinel) drawables.value[1] else top
        this[2] = if (end === drawableSentinel) drawables.value[2] else end
        this[3] = if (bottom === drawableSentinel) drawables.value[3] else bottom
    }
}

// called:

fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
    start: Drawable? = drawableSentinel,
    top: Drawable? = drawableSentinel,
    end: Drawable? = drawableSentinel,
    bottom: Drawable? = drawableSentinel
) {
    val drawables = updateDrawables(start, top, end, bottom, compoundDrawables)
    setCompoundDrawablesWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3])
}
// same for updateCompoundDrawablesRelative with compoundDrawablesRelative etc.
```

Reply:
Corrected : ` Using compoundDrawables 4 times to get the default value will possibly involve creating several arrays (one per call to compoundDrawables` < - getting the arrays only once. Sorry I missed that.